src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -135,6 +135,14 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 
 	CheckCitusVersion(ERROR);
 	EnsureCoordinator();
+
+	/*
+	 * Lock target relation with an exclusive lock - there's no way to make
+	 * sense of this table until we've committed, and we don't want multiple
+	 * backends manipulating this relation.
+	 */
+	Relation relation = relation_open(relationId, ExclusiveLock);
+
 	EnsureTableOwner(relationId);
```

Comment:
PostgreSQL typically takes locks after checking permissions to prevent unprivileged users from taking aggressive locks (e.g. see RangeVarGetRelidExtended). 

Perhaps EnsureTableOwner should be made robust to non-existent tables? (e.g. see RangeVarCallbackForDropRelation)